packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts

```diff
@@ -113,33 +123,38 @@ export class DataSyncMongoDBDataProvider<Type = any> extends MongoDBDataProvider
   }
 
   protected async checkForConflicts(clientData: any, context: GraphbackContext): Promise<ConflictStateMap> {
-    const { idField } = getDatabaseArguments(this.tableMap, clientData);
-    const { fieldNames } = metadataMap;
+    if (this.conflictEngine !== undefined) {
 
-    if (!idField.value) {
-      throw new NoDataError(`Couldn't get document from ${this.collectionName} - missing ID field`)
-    }
+      const queryResult = await this.getServerState(clientData);
+      if (queryResult) {
+        const conflict = this.conflictEngine.checkForConflicts(queryResult, clientData)
```

Comment:
I think while this made sense for the Postgress when we were building it for Mongo this approach makes less sense. There could be some operation happening in the middle of those two. 

Approach here could be that we can research `mongodb optimistic locking`
We should check if we can have conditional errors in mongo - error that is returned if query do not accept some condition. 

While this approach is ok.. it is definitely not production ready as consistency is not guaranteed here.